test/Benchmarks/src/Table/Arithmetic.enso

```diff
@@ -8,45 +8,117 @@ polyglot java import java.lang.Long as Java_Long
+        if i % 10 < 2 then Nothing else i+1
 
-    t = Table.new [["X", x], ["Y", y], ["U", u]]
-
-    assert condition =
-        if condition.not then Panic.throw "Assertion failed"
+    [x, y, u, z]
 
-    assert ((t.at "X" . value_type) == Value_Type.Integer)
-    assert ((t.at "Y" . value_type) == Value_Type.Integer)
-    assert ((t.at "U" . value_type) == Value_Type.Integer)
+create_table : Table
+create_table num_rows =
+    v = create_vectors num_rows
+    x = v.at 0
+    y = v.at 1
+    u = v.at 2
+    z = v.at 3
+
+    t = Table.new [["X", x], ["Y", y], ["U", u], ["Z", z]]
+
+    Runtime.assert ((t.at "X" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "Y" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "U" . value_type) == Value_Type.Integer)
+    Runtime.assert ((t.at "Z" . value_type) == Value_Type.Integer)
     t
 
+foreign arrow int64_new = """
+    new[Int64]
+
+foreign arrow int64_plus = """
+    +[Int64]
 
 type Data
-    Value ~table
+    private Value ~table ~arrow
 
-    create num_rows = Data.Value (create_table num_rows)
+    arrow_plus self = self.arrow.at 0
+    arrow_x self = self.arrow.at 1
+    arrow_y self = self.arrow.at 2
+    arrow_u self = self.arrow.at 3
+    arrow_z self = self.arrow.at 4
+
+    create num_rows = Data.Value (create_table num_rows) <|
+        column_to_arrow v:Vector -> Array =
+            builder = int64_new.new v.length
+            v.map e-> builder.append e
+            builder.build
+
+        v = create_vectors num_rows
+
+        x = column_to_arrow (v.at 0)
+        y = column_to_arrow (v.at 1)
+        u = column_to_arrow (v.at 2)
+        z = column_to_arrow (v.at 3)
+        [int64_plus, x, y, u, z]
 
 
 collect_benches = Bench.build builder->
+    column_arithmetic_plus_fitting d =
+        (d.table.at "X") + (d.table.at "U")
+
+    column_arithmetic_plus_overflowing d =
+        (d.table.at "Y") + (d.table.at "U")
+
+    column_arithmetic_plus_nothing d =
+        (d.table.at "Z") + (d.table.at "U")
+
+    column_arithmetic_multiply_fitting d =
+        (d.table.at "X") * (d.table.at "U")
+
+    column_arithmetic_multiply_overflowing d =
+        (d.table.at "Y") * (d.table.at "U")
+
+    arrow_arithmetic_plus_fitting d =
+        d.arrow_plus d.arrow_x d.arrow_u
+
+    arrow_arithmetic_plus_overflowing d =
+        d.arrow_plus d.arrow_y d.arrow_u
+
+    arrow_arithmetic_plus_nothing d =
+        d.arrow_plus d.arrow_y d.arrow_u
+
     num_rows = 1000000
     data = Data.create num_rows
 
+    Runtime.assert ((column_arithmetic_plus_fitting data . to_vector) == (arrow_arithmetic_plus_fitting data)) "Column and arrow correctness check one"
+    Runtime.assert ((column_arithmetic_plus_overflowing data . to_vector) == (arrow_arithmetic_plus_overflowing data)) "Column and arrow correctness check two"
+    Runtime.assert ((column_arithmetic_plus_nothing data . to_vector) == (arrow_arithmetic_plus_nothing data)) "Column and arrow correctness check three"
```

Comment:
👍

Reply:
Yeah, I really miss ` . should_equal` & co. when writing benchmarks. How are we supposed to know it computes the right values?

Reply:
I guess we could use `should_equal`? It will just throw a panic if it fails.

Reply:
Since #8778 we probably could use `should_equal`. Still I am not sure how to do write such tests properly to avoid:
- initialization overhead
- slowing down the benchmarks

This current `Runtime.assert` relies on the fact that `runEngineDistribution -run test/Benchmarks` runs with _enabled assertions_ and thus it will check the assert. While when running benchmarks on the CI as well as `std-benchmarks/bench` the assertions are disabled and thus this _testing code_ isn't executed at all.

I think it works, but it is a bit fragile.